src/runtime_src/core/common/config_reader.h

```diff
@@ -156,6 +149,24 @@ get_data_transfer_trace()
   return value;
 }
 
+inline std::string
+get_data_transfer_trace_dep_message()
+{
+  static bool emitted = false ;
+  if (!emitted) {
```

Comment:
This is not thread safe, not sure if it matters, probably doesn't?

Reply:
This function is added specifically for a problem we have right now where the OpenCL level device offload and the hardware level device offload are both being loaded and the deprecation message can appear twice.  This happens in a single thread at different times, so thread safety isn't necessary.  Additionally, we are working on delivering a change in 2022.1 where device offload is only loaded at the hardware level once we support device level information on the hardware emulation side without going through the OpenCL level, at which point this function can be removed and the deprecation message moved into the appropriate loaders.